Add HydroGraphNet model and example to PhysicsNeMo

MehdiTaghizadehUVa opened the pull request:
<!-- markdownlint-disable MD013-->
# PhysicsNeMo Pull Request

## Description
<!-- Provide a standalone description of changes in this PR. -->
<!-- Reference any issues closed by this PR with "closes #1234". -->
<!-- Note: The pull request title will be included in the CHANGELOG. -->

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.

## Dependencies

<!-- Call out any new dependencies needed if any -->

mnabian commented on examples/weather/flood_modeling/hydrographnet/outputs/animations/sample_animation.gif:
Could you please move this to (../../../../docs/img)

mnabian commented on physicsnemo/datapipes/gnn/hydrographnet_dataset.py:
```diff
@@ -0,0 +1,865 @@
+        """
+        os.makedirs(os.path.dirname(save_path), exist_ok=True)
+        with open(save_path, 'wb') as f:
+            pickle.dump((self.static_data, self.dynamic_data, self.sample_index), f)
```
Pickle has vulnerability issues and we won't be able to use it in the PhysicsNeMo container. Could you please switch to json?

mnabian commented on physicsnemo/models/layers/__init__.py:
```diff
@@ -55,3 +55,8 @@
+
+from .kan_layers import KolmogorovArnoldNetwork
+
+from .kan_layers import KolmogorovArnoldNetwork
```
Duplicate imports

MehdiTaghizadehUVa commented:
Hi @mnabian,
Thank you for the feedback. I’ve addressed the comments as follows:

Moved sample_animation.gif to docs/img

Removed all functions using pickle due to security concerns

Removed the duplicate import of KolmogorovArnoldNetwork

Please let me know if any further changes are needed.

mnabian commented:
Thanks @MehdiTaghizadehUVa ! Could you please also add unit tests for the datapipe, model, and the KAN layer?

mnabian commented:
/blossom-ci

mnabian commented:
/blossom-ci

mnabian commented:
/blossom-ci

mnabian commented:
/blossom-ci

mnabian commented:
/blossom-ci

mnabian commented:
/blossom-ci

mnabian commented:
/blossom-ci

MehdiTaghizadehUVa commented:
Hi Mohammad,

Thanks for fixing a couple of issues in the code and test files. I’ve now addressed the remaining errors, and everything should be working fine.

Best,
Mehdi

mnabian commented:
/blossom-ci

mnabian commented:
/blossom-ci
